src/Adapter/MailTemplate/MailTemplateDataProvider.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\MailTemplate;
+
+use Language;
+
+/**
+ * This class will provide data from filesystem about MailTemplate.
+ */
+class MailTemplateDataProvider
+{
+    /**
+     * @var Language[]
+     */
+    private $languages;
+
+    public function __construct(array $languages)
+    {
+        $this->languages = $languages;
+    }
+
+    /**
+     * Get all templates.
+     *
+     * @return array Suppliers
+     */
+    public function getTemplates()
+    {
+        $mailsTemplates = [];
+
+        // Mail templates can also be found in the theme folder
+        $themePath = _PS_THEME_DIR_ . 'mails' . DIRECTORY_SEPARATOR;
+        $mailsPath = _PS_ROOT_DIR_ . DIRECTORY_SEPARATOR . 'mails' . DIRECTORY_SEPARATOR;
+
+        foreach ($this->languages as $language) {
+            $languageIsoCode = $language['iso_code'];
+            $languageId = $language['id_lang'];
+
+            $languageMailsPath = $mailsPath . $languageIsoCode;
+            $themeTemplatesDir = $themePath . $languageIsoCode;
+
+            // If there is no folder for the given iso_code in /mails or in /themes/[theme_name]/mails, we bypass this language
+            if (!is_dir($languageMailsPath) && !is_dir($themeTemplatesDir)) {
```

Comment:
```suggestion
            if (!is_dir($languageMailsPath) || !is_dir($themeTemplatesDir)) {
```
Regarding your comment, must be `or` instead of `and`

Reply:
if one of this folders exists, ...
Notice the ! bedore each is_dir

Reply:
Yes, this mean if only one doesn't exist, the condition will return false.
Both directories must exist.

Reply:
Ok, I understood that we need only one of them (at least) to exist
Fixing ...